test/extended/prometheus/prometheus.go

```diff
@@ -278,6 +279,150 @@ var _ = g.Describe("[sig-instrumentation][Late] OpenShift alerting rules", func(
+			"ConfigReloaderSidecarErrors",
+
+			// coredns_dns_responses_total selector may return empty results.
+			"CoreDNSErrorsHigh",
+
+			// cvs_abnormal may be absent.
+			"CsvAbnormalFailedOver2Min",
+			"CsvAbnormalOver30Min",
+
+			// aggregator_unavailable_apiservice_total may be absent.
+			"KubeAggregatedAPIErrors",
+
+			// kube_resourcequota selector may return empty results.
+			"KubeCPUQuotaOvercommit",
+			"KubeMemoryQuotaOvercommit",
+
+			// kube_pod_container_status_waiting_reason may be absent.
+			"KubeContainerWaiting",
+
+			// some metrics may be absent.
+			"KubeDaemonSetRolloutStuck",
+
+			// metrics may be absent when there are no HPA.
+			"KubeHpaMaxedOut",
+			"KubeHpaReplicasMismatch",
+
+			// kube_job_failed may be absent.
+			"KubeJobFailed",
+
+			// kube_node_spec_taint selector may return empty results.
+			"KubeNodeUnreachable",
+
+			// kube_persistentvolumeclaim_access_mode selector may return empty results.
+			"KubePersistentVolumeFillingUp",
+
+			// kube_pod_container_status_waiting_reason selector may return empty results.
+			"KubePodCrashLooping",
+
+			// kube_state_metrics_list_total selector may return empty results.
+			"KubeStateMetricsListErrors",
+			"KubeStateMetricsWatchErrors",
+
+			// mapi_machine_created_timestamp_seconds selector may return empty results.
+			"MachineNotYetDeleted",
+			"MachineWithNoRunningPhase",
+
+			// kube_pod_container_status_last_terminated_reason selector may return empty results.
+			"MultipleContainersOOMKilled",
+
+			// metrics may be absent if RAID is disabled.
+			"NodeRAIDDegraded",
+			"NodeRAIDDiskFailure",
+
+			// metrics may be absent if remote write isn't used.
+			"PrometheusRemoteStorageFailures",
+			"PrometheusRemoteWriteBehind",
+			"PrometheusRemoteWriteDesiredShards",
+
+			// openshift_samples_failed_imagestream_import_info may be absent.
+			"SamplesImagestreamImportFailing",
+			"SamplesRetriesMissingOnImagestreamImportFailing",
+
+			// cluster_operator_conditions selector may return empty results.
+			"SimpleContentAccessNotAvailable",
+
+			// metrics may be absent is there are no tech preview features enabled.
+			"TechPreviewNoUpgrade",
+
+			// thanos query error metrics may be absent.
+			"ThanosQueryGrpcClientErrorRate",
+			"ThanosQueryHttpRequestQueryErrorRateHigh",
+			"ThanosQueryHttpRequestQueryRangeErrorRateHigh",
+			"ThanosSidecarBucketOperationsFailed",
+
+			// cluster_version_available_updates may be empty.
+			"UpdateAvailable",
+		)
+
+		promClient := oc.NewPrometheusClient(context.TODO())
+
+		err := helper.ForEachAlertingRule(alertingRules, func(alert promv1.AlertingRule) sets.String {
+			if exceptions.Has(alert.Name) {
+				return nil
+			}
+
+			// Remove backslashs added by the HTTP API since they can't be parsed by
+			// the promQL parser.
+			query := strings.Replace(alert.Query, "\\", "", -1)
+			expr, err := parser.ParseExpr(query)
+			if err != nil {
+				return sets.NewString("couldn't parse query")
```

Comment:
should we also add embed the query that failed to parse?

Reply:
sure